File: we_8_5_discrete_process_noise_for_a_position.m
% Integrated Navigation Systems — Chapter 8: Optimal Estimation and the Kalman Filter
% Worked Example 8.5 — Discrete process noise for a position velocity model
% Companion code (educational reference implementation; see the book for context).

q=0.5; dt=1;
Qd=q*[dt^3/3 dt^2/2; dt^2/2 dt];
disp(Qd);   % [0.167 0.25; 0.25 0.5]

File: we_8_5_discrete_process_noise_for_a_position.py
# Integrated Navigation Systems — Chapter 8: Optimal Estimation and the Kalman Filter
# Worked Example 8.5 — Discrete process noise for a position velocity model
# Companion code (educational reference implementation; see the book for context).

q,dt=0.5,1
Qd=[[q*dt**3/3,q*dt**2/2],[q*dt**2/2,q*dt]]
print(Qd)   # [[0.167,0.25],[0.25,0.5]]
